### PR TITLE
Run distribution upgrade automatically in some scenarios

### PR DESCRIPTION
**Pull request by DaanDeMeyer**
If we notice extra packages were added by the build script to the
local packages repository or if we're running an incremental build
and there are local packages, do a distribution upgrade so that local
packages can be upgraded without having to throw away the cached
images.

### CODE REVIEW

File: mkosi/installer/apt.py [hunk 1]
```diff
@@ -39,11 +48,21 @@ def setup_apt(context: Context, repos: Sequence[str]) -> None:
...
             for repo in repos:
-                f.write(f"{repo}\n")
+                f.write(
+                    textwrap.dedent(
+                        f"""\
+                        Types: {" ".join(repo.types)}
+                        URIs: {repo.url}
+                        Suites: {repo.suite}
+                        Components: {" ".join(repo.components)}
+                        {"Signed-By" if repo.signedby else "Trusted"}: {repo.signedby or "yes"}
```
Comment (behrmann) [inline mkosi/installer/apt.py:62]:
```suggestion
                        {f"Signed-By: {repo.signedby}" if repo.signedby else "Trusted: yes"}
```
contra: nested f-string
pro: easy to read what belongs where, tested to wrk on 3.9

File: mkosi/installer/apt.py [hunk 2]
```diff
@@ -39,11 +48,21 @@ def setup_apt(context: Context, repos: Sequence[str]) -> None:
...
+                        """
```
Comment (behrmann) [inline mkosi/installer/apt.py:63]: The whole thing would be great as the `__str__` method on `AptRepository`—would save the `textwrap.dedent`, too.

**Review by behrmann — APPROVED** [review]
Maybe we could make a function on context
```
def need_local_repo(self) -> bool:
    return any(self.packages.iterdir())
```
because somehow I stumbled over this twice when reading it and thought about packages to be installed and not the package dir.

**Comment by DaanDeMeyer** [comment]
Will submit a new PR

**Pull request merged**